### PR TITLE
Retrying subscribe/unsubscribe operations immediately in case of exceptions

### PR DESCRIPTION
**Pull request by SzymonPobiega**
Also using `TABLOCK` hint in the subscribe query

### CODE REVIEW

**Comment by SzymonPobiega** [comment]
@SimonCropp @weralabaj how about we change it to a normal `if exists` query using `serializable` isolation level. I guess it would be easier to understand than the overcomplicated syntax of `merge`

**Comment by weralabaj** [comment]
@SzymonPobiega I wonder if actually locking level is an issue here really. I mean the behavior was as desired, timeout subscriptions were processed, so I think ensuring proper logging level (info probably is too high for internal stuff that shouldn't concern users) and using dedicated retries instead of FLR should be a satisfactory solution. What do you think?

**Comment by SzymonPobiega** [comment]
Yeah. Also, I was not able to reproduce this behavior in my tests :(

**Comment by weralabaj** [comment]
So maybe we are a bit overzealous with the changes? Happy to get more opinions, but I see in the related case you remarked that logged error messages that are below "error" level are automatically resolved and according to our philosophy should not be a problem. And that makes sense IMO, especially in this case where potential deadlocks occur only at endpoint startup.

**Comment by weralabaj** [comment]
@Particular/sqlpersistence-maintainers your input in the discussion would be appreciated.

**Comment by MarcinHoppe** [comment]
This change has been beneficial for NH persistence in the past so I'm 👍 to include it here as well.

**Comment by SzymonPobiega** [comment]
I'll rebase this.

File: src/SqlPersistence/Subscription/SubscriptionPersister.cs [hunk 1]
```diff
@@ -32,32 +32,61 @@ public SubscriptionPersister(Func<DbConnection> connectionBuilder, string tableP
...
+        var attempts = 0;
+        while (true)
+        {
+            try
+            {
+                await action().ConfigureAwait(false);
+                return;
+            }
+            catch (Exception)
+            {
```
Comment (SzymonPobiega) [inline src/SqlPersistence/Subscription/SubscriptionPersister.cs]: Yep. Done.

**Comment by SzymonPobiega** [comment]
@weralabaj you can press the big green button when it finishes building :-)

**Pull request merged**